docs/dev/pagination-v2.md

```diff
@@ -0,0 +1,173 @@
+# Pagination in v2 Engine
+
+<!-- Why is pagination necessary? -->
+Pagination allows a SQL plugin client to retrieve arbitrarily large
+results sets one subset at a time.
+
+A cursor is a SQL abstraction for pagination. A client can open a cursor, retrieve a subset of data given a cursor and close a cursor.
+<!-- What applications will use pagination? -->
+
+Currently, SQL plugin does not provide SQL cursor syntax. However, the SQL REST endpoint can return result a page at a time.
+
+JDBC and ODBC drivers are the main users of cursor REST API.
+
+<!-- How will applications use pagination? -->
+
+# Scope
+<!-- What scenarios are excluded from this discussion? Why? -->
+
+The primary goal is to reach feature parity with the v1 engine. v1 engine suppors pagination only for simple `SELECT` queries with optional `WHERE` or `ORDER BY` clauses.
+
+## Compatibility with v1 engine
```

Comment:
We should have a schedule section too.  
Phase I (must haves) with a list of 'tasks'/'features' and github tickets that will get solved by this. 
Phase II (should haves) with a list of 'tasks'/'features' and github tickets that describe additional necessary pagination API features that are really needed for the experience. 
Phase III or (future work) with a list of items that are for the future.